File: MicroBenchmarks/Eratosthenes-Sieve.js
var thePrimes = [];

function getPrimesBelow(n)
{
  if (n <= 2) return;
  var prime = new Array(n);
  for (var i = 2; i < n; i++) prime[i] = true;
  prime[1] = prime[0] = false;
  for (var i = 2; i * i < n; i++)
  {
    if (prime[i]) for (var j = i; j * i < n; j++) prime[j * i] = false;
  }
  for (var i = 0; i < prime.length; i++) if (prime[i]) thePrimes.push(i);
}

var N = 15485865;

WScript.StdOut.WriteLine("MS JScript... About to find all primes below " + N.toString() + "...")
WScript.StdOut.WriteLine("(Press return)");
WScript.StdIn.ReadLine();
var t0 = new Date().valueOf();
getPrimesBelow(N);
var elapsed = new Date().valueOf() - t0;
WScript.StdOut.WriteLine(thePrimes.length.toString() + " primes found in " + elapsed.toString() + " ms...");
WScript.StdOut.WriteLine("first... " + thePrimes[0].toString());
WScript.StdOut.WriteLine("last ... " + thePrimes[thePrimes.length - 1].toString());
WScript.StdOut.WriteLine("(Press return to exit)");
WScript.StdIn.ReadLine();

File: MicroBenchmarks/Eratosthenes-Sieve.py
from time import time

def started_time():
	return int(1000 * time())

def elapsed_time(start):
	return int(1000 * time()) - start

thePrimes = []

def getPrimesBelow(n):
	if (n <= 2):
		return
	prime = [True] * n
	prime[0] = False
	prime[1] = False
	i = 2
	while (i * i < n):
		if (prime[i]):
			j = i
			while (j * i < n):
				prime[j * i] = False
				j = j + 1
		i = i + 1
	for i in range(0, len(prime)):
		if (prime[i]):
			thePrimes.append(i)

N = 15485865

print("Python... About to find all primes below " + str(N) + "...")
print("(Press return)")
input()

t0 = started_time()
getPrimesBelow(N)
elapsed = elapsed_time(t0)
print(str(len(thePrimes)) + " primes found in " + str(elapsed) + " ms...")
print("first... " + str(thePrimes[0]));
print("last ... " + str(thePrimes[len(thePrimes) - 1]));
print("(Press return to exit)")
input()
